extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -621,15 +627,16 @@ export class AmpDatePicker extends AMP.BaseElement {
    */
   getHiddenInputId_(form, type) {
     const id = this.element.id;
+    const name = DateFieldNameByType[type];
     if (form) {
-      const alternativeName = `${id}-${type}`;
-      if (!form.elements[type]) {
-        return type;
+      const alternativeName = `${id}-${name}`;
+      if (!form.elements[name]) {
+        return name;
       } else if (id && !form.elements[alternativeName]) {
-        return `${id}-${type}`;
+        return alternativeName;
```

Comment:
Nit: no need for else-ifs, just short-circuit.

Reply:
Nice, will do

Reply:
Well, the build is green now. I think I'll merge now and add this in a follow up PR 😄